src/store/media-store.ts

```diff
@@ -222,11 +230,66 @@ export const createSettersSlice: StateCreator<
 
 	_handleProgress: (currentTime: number) =>
 		set(state => {
+			let shouldRunTimeAlarm = false;
 			const currentRelativeTime = Math.min(
 				state.endTime,
 				Math.max(0, currentTime - state.startTime),
 			);
+			let newTimeAlarm = {
+				current: state.currentTimeAlarm || 0,
+				next: state.nextTimeAlarm || 0,
+			};
+			const [timeAlarm, timeBeforeAlarm] = [
+				state.timeAlarm,
+				state.timeBeforeAlarm || 0,
+			];
+
+			// Creating and updating `onTimeAlarm`
+			if (timeAlarm && timeAlarm.length > 0) {
+				// Time that was played previously
+				const previousTime = state.currentTime;
+				// Get the initial state
+				const isInitialized =
+					previousTime === state.currentTimeAlarm &&
+					previousTime === state.nextTimeAlarm;
+				// Run 1 search for 2 consecutive values
+				if (
+					(newTimeAlarm.next < currentRelativeTime &&
```

Comment:
Can `newTimeAlarm.current < currentRelativeTime` not be true if `newTimeAlarm.next < currentRelativeTime` is true? I.e. can the current alarm be after the next alarm? => If not: checking for the next alarm should be sufficient.
[If yes: pls add a comment to the code why this is]

Reply:
We keep 2 values of the alarm(current and next).
And if the currentRelativeTime(the defacto time) is between this interval [next, current] - we do not to run any useless searches.
The search for these 2 values will be only when we passed this interval